src/main/java/duke/parse/Parser.java

```diff
@@ -0,0 +1,195 @@
+            }
+            newSplitInput[1] = fullTaskDetails;
+
+        } else {
+            String dateDelimiter = "";
+            if (action.equals("event")) {
+                dateDelimiter = " /at ";
+            } else if (action.equals("deadline")) {
+                dateDelimiter = " /by ";
+            }
+
+            String[] inputArr = fullTaskDetails.split(dateDelimiter);
+
+            if (!(inputArr.length == 2)) {
+                // inputArr.length should be 2 to indicate proper formatting of the command and
+                // necessary descriptions.
+                throw new DukeException(
+                        "Your " + action + " command does not follow proper formatting!\n"
+                        + "A " + action + " command should follow this convention:\n"
+                        + action + " <task description> " + dateDelimiter + " <date description> <3");
+            }
+
+            newSplitInput[1] = inputArr[0];
+            newSplitInput[2] = inputArr[1];
+
+        }
+        return newSplitInput;
+    }
+
+    /**
+     * Performs actions based on the input.
+     * Commands:
+     *   - 'Bye': Terminates the programme.
+     *   - 'list': Lists the current stored tasks.
+     *   - 'mark x', where x is a valid task index: Mark task x as done.
+     *   - 'unmark x', where x is a valid task index: Mark task x as undone.
+     *   - 'todo *', where * refers to any input: Create a Todo task.
+     *   - 'event x /at y', where x and y refers to any input: Create an Event task that will happen at y.
+     *   - 'deadline x /by y', where x and y refers to any input: Create a Deadline task that is due by y.
+     *
+     * @param input The input given by the user.
+     * @param tasks The TaskList to perform appropriate actions on, after
+     *              parsing the input.
+     * @return Returns true if the programme should continue prompting the user
+     *         for inputs. Returns false if the programme is to be terminated.
+     * @throws DukeException When there are invalid inputs.
+     */
+    public static boolean settleInput(String input, TaskList tasks) throws DukeException {
+        String[] splitInput = input.split(" ");
+        String action = splitInput[0];
+        if (action.equals("Bye")) {
+            Command command = new ByeCommand(splitInput, tasks);
+            return command.performAction();
+
+        } else if (action.equals("list")) {
+            // Throw error if the input contains anything other than 'list'
+            if (!(splitInput.length == 1)) {
+                throw new DukeException("</3 your formatting for the list command is wrong - please just type list!");
+            }
+            Command command = new ListCommand(splitInput, tasks);
+            return command.performAction();
+
+        } else if (
+                action.equals("mark")
+                || action.equals("unmark")
```

Comment:
Perhaps you can group mark, unmark and delete error together to shorten the method? Since they are checking the same conditions.